extensions/interactions/GraphInput/GraphInput.js

```diff
@@ -583,6 +583,86 @@ oppia.directive('graphViz', function() {
 
 oppia.factory('graphUtilsService', [function() {
   return {
+    GRAPH_ADJACENCY_MODE: {
+      DIRECTED: 'directed',
+      INVERTED: 'inverted',
+      UNDIRECTED: 'undirected'
+    },
+    DFS_STATUS: {
+      VISITED: 'visited',
+      UNVISITED: 'unvisited',
+      STILL_VISITING: 'still visiting'
+    },
+    /**
+     * @param {object} graph A graph object.
+     * @param {string} mode A string indicating the mode.
+     * @return {array} An adjacency list. Depending on the mode, the list has
+     *   all edges (directed),
+     *   all edges inverted (inverted),
+     *   or all edges in both directions, as though the graph were undirected
+     *   (undirected)
+     */
+    constructAdjacencyLists: function(graph, mode) {
```

Comment:
Is mode here the mode of the output list? If so I would rename it to something more explicit.